include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -4156,10 +4442,52 @@ ::submit_gradient (const Tensor<1,n_components_,
 
 
 
-template <int dim, int n_components_, typename Number>
+template <int dim, int n_components_, typename Number, bool is_face>
+inline DEAL_II_ALWAYS_INLINE
+void
+FEEvaluationBase<dim,n_components_,Number,is_face>
+::submit_normal_gradient (const Tensor<1,n_components_,VectorizedArray<Number> > grad_in,
+                          const unsigned int q_point)
+{
+#ifdef DEBUG
+  AssertIndexRange (q_point, this->n_quadrature_points);
+  this->gradients_quad_submitted = true;
+  Assert (this->normal_x_jacobian != nullptr, ExcNotInitialized());
+#endif
+
+  if (this->cell_type == internal::MatrixFreeFunctions::cartesian)
+    for (unsigned int comp=0; comp<n_components; comp++)
+      {
+        for (unsigned int d=0; d<dim-1; ++d)
+          this->gradients_quad[comp][d][q_point] = VectorizedArray<Number>();
+        this->gradients_quad[comp][dim-1][q_point] =
+          grad_in[comp] * (this->normal_x_jacobian[0][dim-1] *
+                           this->J_value[0] * this->quadrature_weights[q_point]);
+      }
+  else
+    {
+      const unsigned int index =
+        this->cell_type <= internal::MatrixFreeFunctions::affine ? 0 : q_point;
+      for (unsigned int comp=0; comp<n_components; comp++)
+        {
+          VectorizedArray<Number> factor = grad_in[comp] *
+                                           this->J_value[index];
+          if (this->cell_type <= internal::MatrixFreeFunctions::affine)
+            factor = factor * this->quadrature_weights[q_point];
```

Comment:
aren't there `factor *= ...`?

Reply:
The problem is that `quadrature_weight` is of type `Number` whereas `factor` is `VectorizedArray<Number>`, for which we do only have `operator*` but not `operator*=`. I guess we should extend this at some point, but again this is a bit too much so close to the release as it is in 7 classes (base, 2x SSE, 2x AVX, 2x AVX-512). If you want to keep track, please I can open an issue.

Reply:
sure, i was just curios whether this can be done with the current state.

Since that's not the case, just disregard my comment.